### PR TITLE
feat: getStandardKcat use custom list of non-enzyme reactions to ignore

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
- feat:
  - `getStandardKcat` can use `data/pseudoRxns.tsv` (if it exists) to define custom non-enzyme reactions that should be ignored when assigning standard kcats. (solves #280)
  - `getStandardKcat` removes previous standard kcat assignments, so that the function can be run repeatedly on the same model.
- fix:
  - `getStandardKcat` also specifies `model.ec.source{i} = 'standard'` if a standard kcat is assigned to a zero-kcat reaction. (solves #286)
  - remaining corrections for prot_pool_exchange minimization (solves #287)

**I hereby confirm that I have:**

- [x] Selected `develop` as a target branch (top left drop-down menu)

### CODE REVIEW

**Review by ae-tafur — APPROVED** [review]
Looks good!

File: protocol.m [hunk 1]
```diff
@@ -130,8 +130,25 @@
 ecModel = getKcatAcrossIsoenzymes(ecModel);
 
 % STEP 12 Get standard kcat
-% Assign an enzyme cost to reactions without gene assocation (except
-% exchange, transport and pseudoreactions)
+% Assign an enzyme cost to reactions without gene assocation. These
```
Comment (Yu-sysbio) [inline protocol.m:133]: Maybe change `an enzyme cost` to `a protein cost` to keep consistent with the manuscript.

File: protocol.m [hunk 2]
```diff
@@ -130,8 +130,25 @@
...
+% The following reactions are exempted:
+% A Exchange reactions: exchanging a metabolite across the model boundary,
+%   not representing a real enzymatic reaction.
+% B Transport reactions: transporting a metabolite with the same name from
+%   one compartment to another. Real transport reactions should already be
+%   annotated with grRules, so that the remaining non-annotated reactions
+%   are mostly representing diffusion or pseudotransport processes such as
+%   vesicles moving from ER to Golgi. While proteins are involved in such
+%   processes, they are not catalyzed by enzymes.
+% C Pseudoreactions: any other reaction that should be considered to be
```
Comment (Yu-sysbio) [inline protocol.m:144]: `should be considered` -> `should not be considered` ?

File: src/geckomat/gather_kcats/getStandardKcat.m [hunk 3]
```diff
@@ -126,69 +133,88 @@
...
 
+% Get custom list of reaction IDs to ignore, if existing. First column
+% contains reaction IDs, second column contains reaction names for
+% reference only.
+if exist(fullfile(params.path,'data','pseudoRxns.tsv'),'file')
+    fID        = fopen(fullfile(params.path,'data','pseudoRxns.tsv'));
+    fileData   = textscan(fID,'%s %s','delimiter','\t');
+    fclose(fID);
+    customRxns = fileData{1};
+    customRxns = find(strcmp(customRxns,model.rxns));
```
Comment (Yu-sysbio) [inline src/geckomat/gather_kcats/getStandardKcat.m:144]: Error using `strcmp`
Inputs must be the same size or either one can be a scalar.

**Review by Yu-sysbio — COMMENTED** [review]
Looks good!

File: protocol.m (hunk 1 above)
Comment (edkerk) [reply protocol.m:133]:
```suggestion
% Assign a protein cost to reactions without gene assocation. These
```

File: src/geckomat/gather_kcats/getStandardKcat.m (hunk 3 above)
Comment (edkerk) [reply src/geckomat/gather_kcats/getStandardKcat.m:144]:
```suggestion
    customRxns = find(ismember(model.rxns,customRxns));
```

**Pull request merged**